### PR TITLE
N_ell covariance marginalization

### LINKED ISSUE #31: cov: Add noise marginalization
Each mapper should have an option `marginalize_noise`. If `True`, the covariance has to incorporate this effect.

### PR DESCRIPTION
**Pull request by damonge**
Closes #31

### CODE REVIEW

File: xcell/cls/cov.py [hunk 1]
```diff
@@ -105,6 +111,10 @@ def get_covariance(self):
                                       cla1b1, cla1b2, cla2b1, cla2b2,
                                       wa, wb)
 
+        if self.nl_marg:
+            _, nl = self.clA1A2.get_ell_nl
```
Comment (carlosggarcia) [inline xcell/cls/cov.py:115]: You need to call the function (i.e. add `()` to `.get_ell_nl`)

File: xcell/cls/cov.py [hunk 2]
```diff
@@ -105,6 +111,10 @@ def get_covariance(self):
...
+            cov += self.nl_prior**2 * (nl[:, None] * nl[None, :])
```
Comment (carlosggarcia) [inline xcell/cls/cov.py:116]: Will this work? This will have 3 axis instead of two. `nl.shape = (dof, number_of_bpw)`

**Review by carlosggarcia — CHANGES_REQUESTED** [review]

**Comment by damonge** [comment]
alright @carlosggarcia, ready for review

File: xcell/tests/test_cls.py [hunk 3]
```diff
@@ -64,6 +64,19 @@ def test_smoke():
...
+    data = get_config(0.2)
+    data['tracers']['Dummy__0']['nl_marginalize'] = True
+    data['tracers']['Dummy__0']['nl_prior'] = 1E30
+    data['tracers']['Dummy__0']['noise_level'] = 1E-5
+    cov_class = Cov(data, 'Dummy__0', 'Dummy__0', 'Dummy__0', 'Dummy__0')
+    cov = cov_class.get_covariance()
+    num_l = len(cov)
+    oo = np.ones(num_l)
+    chi2 = np.dot(oo, np.linalg.solve(cov, oo))
+    assert np.fabs(chi2) < 1E-5*num_l
```
Comment (carlosggarcia) [inline xcell/tests/test_cls.py:77]: Why `<1 E-5 * num_l`?

**Review by carlosggarcia — COMMENTED** [review]
Just one question, for the record

File: xcell/tests/test_cls.py (hunk 3 above)
Comment (damonge) [reply xcell/tests/test_cls.py:77]: this is the chi2 with respect to a theory that is equivalent to the noise you just marginalized over, so it should be zero

**Review by carlosggarcia — APPROVED** [review]

**Pull request merged**